docs/decisions/000x-internal-editor-testability-decisions.md

```diff
@@ -0,0 +1,136 @@
+# Internal editor testability decision
+
+# Increased complexity for the sake of testability
+The internally-managed editors in this repo (as of now planned to include text, video, and problem types) follow a number of patterns that increase the complexity of parts of the code slightly, in favor of providing increased testability around their behavior.
```

Comment:
I am not sure that idea makes sense. What usually happens is that hard-to-test components are relatively complex, and reducing that complexity makes them easy to test. If you need to increase code complexity to test better, it is more likely a sign that the code is already too complex (and then you make matters worse.) I know testing is hard but the better way is to decouple your code enough that it's testable, develop better testing tools, and also testing the right things (I recommend starting to use react-testing-library to make your life easier here.)